Notifications bar

lontivero opened the pull request:
This PR add a notifications bar on top of the status bar. It simplifies and reduces a lot the effort required to display warnings, errors or other info in Wasabi (take a look at the code and you will see that most of the changes are for removing code in the existing viewmodels and views). See some screenshorts in the issue https://github.com/zkSNACKs/WalletWasabi/issues/845

It is not perfect yet and it could required some polish yet but I would like to have some feedback first. @danwalmsley @molnard could you share your thoughts, please?

nopara73 commented:
I'm still voting for Toast Notifications, so if you want to get this PR in it must be beautiful and perfect 😄

lontivero commented:
I know, that's why it is beautiful and perfect. I've also had in mind the fact that we will have toast notifications soon. This PR improves the way Wasabi provides visual feedback to the users.

lontivero commented:
Rebased.

lontivero commented:
This is a extremely low quality gif image just to let you see how the notification bar looks like:

![out](https://user-images.githubusercontent.com/127973/54882133-ab8eb380-4e35-11e9-8e8a-d7a234b30890.gif)

nopara73 commented:
I it test-ready?

nopara73 commented:
Anyway, tested. Issues I found:

1. Contrast is wrong.
2. There's a bug when the notification starts.
3. Multiple clicking is buggy.
4. When the user hovers the mouse to the band the band should not disappear.
5. Error messages are sometimes not what we expect. The user should be able to copy it.

lontivero commented:
Ack. I will implement those requirements then.

lontivero commented on WalletWasabi.Gui/Global.cs:
```diff
@@ -302,6 +305,35 @@ public static void InitializeNoWallet()
+				arguments= $"--expire-time=3000 \"Wasabi\" \"{e.NotificationText}\"";
+			}
+
+			using (var process = Process.Start(new ProcessStartInfo
```
fixed

lontivero commented on WalletWasabi.Gui/Controls/NotificationBar.xaml:
```diff
@@ -0,0 +1,26 @@
+    <DockPanel LastChildFill="True" Margin="4 0" HorizontalAlignment="Center">
+      <Panel>
+        <StackPanel Orientation="Horizontal" Spacing="4">
+          <TextBlock Text="{Binding NotificationText}" VerticalAlignment="Center"/>
```
I tried bold but the text doesnt read well

lontivero commented on WalletWasabi.Gui/ViewModels/NotificationBarViewModel.cs:
```diff
@@ -0,0 +1,81 @@
+			Global.NotificationManager.NotificationAvailable += (s, n) => _notificationQueue.Enqueue(n);
+			_timer = new DispatcherTimer
+			{
+				Interval = TimeSpan.FromSeconds(3)
```
Now it is 7 seconds opened and almost a second fading out.

lontivero commented:
I've implemented the requested changes:

> Contrast is wrong.

Now it has more contracts but given it is not defined anywhere it can be still wrong.

> There's a bug when the notification starts.

I've rewritten the thing so I don't remember this issue.

>  Multiple clicking is buggy, newer notifications should come on above the one below if it did not yet disappeared.

This is done. To prevent a long list of notifications in the UI duplicated notifications happening in a short period of time are marked as duplicated.

> When the user hovers the mouse to the band the band should not disappear.

I doesn't disappear now.

>Error messages are sometimes not what we expect. The user should be able to copy it.

Users can copy the message now.

Error found by @molnard was fixed too.

lontivero commented:
This PR works okay but the @danwalmsley solution is clearly superior so, I will close it.

nopara73 commented:
The bulk of this work is the when and what notifications should we add so it'll be still valuable, maybe we could even start out as a fork of this.
